src/liballoc/heap.rs

```diff
@@ -16,6 +16,18 @@
 
 use core::{isize, usize};
 
+#[allow(improper_ctypes)]
+extern {
+    #[allocator]
```

Comment:
This crate has `#![feature(allocator)]` at the top, is that what you're looking for?

This attribute existed before this PR and is a way to [add an attribute](https://github.com/rust-lang/rust/blob/master/src/librustc_trans/trans/attributes.rs#L119-L121) to the function for optimization purposes.